### ISSUE TITLE
build fails with ghc-9.0.x

### ISSUE THREAD

**Issue by mimi1vx** [fpco/th-utilities#14]
caused by https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0099-explicit-specificity.rst

```
[   61s] + ./Setup configure --prefix=/usr --libdir=/usr/lib64 --docdir=/usr/share/licenses/ghc-th-utilities '--libsubdir=$compiler/$pkgid' '--datasubdir=$pkgid' --ghc '--dynlibdir=/usr/lib64/$compiler/$pkgid' --global '--ghc-options= -optc-O2 -optc-Wall -optc-D_FORTIFY_SOURCE=2 -optc-fstack-protector-strong -optc-funwind-tables -optc-fasynchronous-unwind-tables -optc-fstack-clash-protection -optc-Werror=return-type ' -v -p --enable-shared --htmldir=/usr/share/doc/packages/ghc/html/libraries/th-utilities-0.2.4.1 --global
[   62s] Using Parsec parser
[   62s] Configuring th-utilities-0.2.4.1...
[   62s] Dependency base >=4.7 && <5: using base-4.15.0.0
[   62s] Dependency bytestring: using bytestring-0.10.12.1
[   62s] Dependency containers: using containers-0.6.4.1
[   62s] Dependency directory: using directory-1.3.6.1
[   62s] Dependency filepath: using filepath-1.4.2.1
[   62s] Dependency primitive: using primitive-0.7.1.0
[   62s] Dependency syb: using syb-0.7.2.1
[   62s] Dependency template-haskell >=2.7: using template-haskell-2.17.0.0
[   62s] Dependency text: using text-1.2.4.1
[   62s] Dependency th-orphans: using th-orphans-0.13.11
[   62s] Source component graph: component lib
[   62s] Configured component graph:
[   62s]     component th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX
[   62s]         include base-4.15.0.0
[   62s]         include bytestring-0.10.12.1
[   62s]         include containers-0.6.4.1
[   62s]         include directory-1.3.6.1
[   62s]         include filepath-1.4.2.1
[   62s]         include primitive-0.7.1.0-AXiSfxjYwi4FGeUoHsPra4
[   62s]         include syb-0.7.2.1-1KAKIfi7jVfHb7rx2ko4wD
[   62s]         include template-haskell-2.17.0.0
[   62s]         include text-1.2.4.1
[   62s]         include th-orphans-0.13.11-rwTnh8qdhLAEKdXc3pbo3
[   62s] Linked component graph:
[   62s]     unit th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX
[   62s]         include base-4.15.0.0
[   62s]         include bytestring-0.10.12.1
[   62s]         include containers-0.6.4.1
[   62s]         include directory-1.3.6.1
[   62s]         include filepath-1.4.2.1
[   62s]         include primitive-0.7.1.0-AXiSfxjYwi4FGeUoHsPra4
[   62s]         include syb-0.7.2.1-1KAKIfi7jVfHb7rx2ko4wD
[   62s]         include template-haskell-2.17.0.0
[   62s]         include text-1.2.4.1
[   62s]         include th-orphans-0.13.11-rwTnh8qdhLAEKdXc3pbo3
[   62s]         TH.Derive=th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX:TH.Derive,TH.Derive.Storable=th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX:TH.Derive.Storable,TH.ReifySimple=th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX:TH.ReifySimple,TH.RelativePaths=th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX:TH.RelativePaths,TH.Utilities=th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX:TH.Utilities
[   62s] Ready component graph:
[   62s]     definite th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX
[   62s]         depends base-4.15.0.0
[   62s]         depends bytestring-0.10.12.1
[   62s]         depends containers-0.6.4.1
[   62s]         depends directory-1.3.6.1
[   62s]         depends filepath-1.4.2.1
[   62s]         depends primitive-0.7.1.0-AXiSfxjYwi4FGeUoHsPra4
[   62s]         depends syb-0.7.2.1-1KAKIfi7jVfHb7rx2ko4wD
[   62s]         depends template-haskell-2.17.0.0
[   62s]         depends text-1.2.4.1
[   62s]         depends th-orphans-0.13.11-rwTnh8qdhLAEKdXc3pbo3
[   62s] Using Cabal-3.4.0.0 compiled by ghc-9.0
[   62s] Using compiler: ghc-9.0.1
[   62s] Using install prefix: /usr
[   62s] Executables installed in: /usr/bin
[   62s] Libraries installed in: /usr/lib64/ghc-9.0.1/th-utilities-0.2.4.1
[   62s] Dynamic Libraries installed in: /usr/lib64/ghc-9.0.1/th-utilities-0.2.4.1
[   62s] Private executables installed in:
[   62s] /usr/libexec/x86_64-linux-ghc-9.0.1/th-utilities-0.2.4.1
[   62s] Data files installed in: /usr/share/th-utilities-0.2.4.1
[   62s] Documentation installed in: /usr/share/licenses/ghc-th-utilities
[   62s] Configuration files installed in: /usr/etc
[   62s] No alex found
[   62s] Using ar found on system at: /usr/bin/ar
[   62s] No c2hs found
[   62s] No cpphs found
[   62s] No doctest found
[   62s] Using gcc version 10 found on system at: /usr/bin/gcc
[   62s] Using ghc version 9.0.1 found on system at: /usr/bin/ghc
[   62s] Using ghc-pkg version 9.0.1 found on system at: /usr/bin/ghc-pkg
[   62s] No ghcjs found
[   62s] No ghcjs-pkg found
[   62s] No greencard found
[   62s] Using haddock version 2.24.0 found on system at: /usr/bin/haddock
[   62s] No happy found
[   62s] Using haskell-suite found on system at: haskell-suite-dummy-location
[   62s] Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
[   62s] No hmake found
[   62s] Using hpc version 0.68 found on system at: /usr/bin/hpc
[   62s] Using hsc2hs version 0.68.7 found on system at: /usr/bin/hsc2hs
[   62s] No hscolour found
[   62s] No jhc found
[   62s] Using ld found on system at: /usr/bin/ld
[   62s] Using pkg-config version 1.7.3 found on system at: /usr/bin/pkg-config
[   62s] Using runghc version 9.0.1 found on system at: /usr/bin/runghc
[   62s] Using strip version 2.36 found on system at: /usr/bin/strip
[   62s] Using tar found on system at: /usr/bin/tar
[   62s] No uhc found
[   62s] + '[' -x Setup ']'
[   62s] + LANG=en_US.utf8
[   62s] + ./Setup build -v
[   62s] Component build order: library
[   62s] /usr/bin/ghc-pkg init dist/package.conf.inplace
[   62s] creating dist/build
[   62s] creating dist/build/autogen
[   62s] creating dist/build/autogen
[   62s] Preprocessing library for th-utilities-0.2.4.1..
[   62s] Building library for th-utilities-0.2.4.1..
[   62s] creating dist/build
[   62s] /usr/bin/ghc --make -fbuilding-cabal-package -O -static -dynamic-too -dynosuf dyn_o -dynhisuf dyn_hi -outputdir dist/build -odir dist/build -hidir dist/build -stubdir dist/build -i -idist/build -isrc -idist/build/autogen -idist/build/global-autogen -Idist/build/autogen -Idist/build/global-autogen -Idist/build -optP-include -optPdist/build/autogen/cabal_macros.h -this-unit-id th-utilities-0.2.4.1-DJc7wb4GWbG9fC08PgzwxX -hide-all-packages -Wmissing-home-modules -no-user-package-db -package-db dist/package.conf.inplace -package-id base-4.15.0.0 -package-id bytestring-0.10.12.1 -package-id containers-0.6.4.1 -package-id directory-1.3.6.1 -package-id filepath-1.4.2.1 -package-id primitive-0.7.1.0-AXiSfxjYwi4FGeUoHsPra4 -package-id syb-0.7.2.1-1KAKIfi7jVfHb7rx2ko4wD -package-id template-haskell-2.17.0.0 -package-id text-1.2.4.1 -package-id th-orphans-0.13.11-rwTnh8qdhLAEKdXc3pbo3 -XHaskell2010 TH.Derive TH.Derive.Storable TH.ReifySimple TH.RelativePaths TH.Utilities TH.Derive.Internal -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates -optc-O2 -optc-Wall '-optc-D_FORTIFY_SOURCE=2' -optc-fstack-protector-strong -optc-funwind-tables -optc-fasynchronous-unwind-tables -optc-fstack-clash-protection '-optc-Werror=return-type'
[   63s] [1 of 6] Compiling TH.Derive.Internal ( src/TH/Derive/Internal.hs, dist/build/TH/Derive/Internal.o, dist/build/TH/Derive/Internal.dyn_o )
[   63s] [2 of 6] Compiling TH.RelativePaths ( src/TH/RelativePaths.hs, dist/build/TH/RelativePaths.o, dist/build/TH/RelativePaths.dyn_o )
[   64s] [3 of 6] Compiling TH.Utilities     ( src/TH/Utilities.hs, dist/build/TH/Utilities.o, dist/build/TH/Utilities.dyn_o )
[   64s] 
[   64s] src/TH/Utilities.hs:18:18: error:
[   64s]     • Expecting one more argument to ‘TyVarBndr’
[   64s]       Expected a type, but ‘TyVarBndr’ has kind ‘* -> *’
[   64s]     • In the type signature: tyVarBndrName :: TyVarBndr -> Name
[   64s]    |
[   64s] 18 | tyVarBndrName :: TyVarBndr -> Name
[   64s]    |                  ^^^^^^^^^
```

**Comment by mgsloan** [comment]
Thanks for reporting!  Fixed in version `0.2.4.2`

**Comment by Bodigrim** [comment]
As a Hackage trustee I revised earlier releases with `template-haskell < 2.17`.
